lib/devices/ios/ios.js

```diff
@@ -1190,8 +1148,16 @@ IOS.prototype.prelaunchSimulator = function (cb) {
     return cb();
   }
 
+  var xcodeFolder = null;
+  xcode.getPath(function (err, xcodePath) {
+    if (err) {
+      return cb(new Error('Could not find xcode folder. Needed to start simulator. ' + err.message));
+    }
+    xcodeFolder = xcodePath;
+  });
+
   logger.debug("Pre-launching simulator");
-  var iosSimPath = path.resolve(this.xcodeFolder,
+  var iosSimPath = path.resolve(xcodeFolder,
```

Comment:
xcodeFolder will almost always be incorrectly `null` here, right? since `xcode.getPath` is async?